solidity/contracts/converter/ConverterBase.sol

```diff
@@ -587,4 +669,126 @@ contract ConverterBase is IConverter, TokenHandler, TokenHolder, ContractRegistr
+	 *
+	 * @param newController new feesController
+	 */
+	function setFeesController(address newController) external onlyOwner {
+		require(newController != address(0), "ERR_ZERO_ADDRESS");
+		address oldController = feesController;
+		feesController = newController;
+
+		emit SetFeesController(msg.sender, oldController, newController);
+	}
+
+	/**
+	 * @notice Set the wrBTC contract address.
+	 *
+	 * @param newWrbtcAddress The address of the wrBTC contract.
+	 * */
+	function setWrbtcAddress(address newWrbtcAddress) external onlyOwner {
+		require(newWrbtcAddress != address(0), "ERR_ZERO_ADDRESS");
+		address oldwrbtcAddress = address(wrbtcAddress);
+		wrbtcAddress = newWrbtcAddress;
+
+		emit SetWrbtcAddress(msg.sender, oldwrbtcAddress, newWrbtcAddress);
+	}
+
+	/**
+	 * @notice Set the SOVToken contract address.
+	 *
+	 * @param newSOVTokenAddress The address of the SOV Token contract.
+	 * */
+	function setSOVTokenAddress(address newSOVTokenAddress) external onlyOwner {
+		require(newSOVTokenAddress != address(0), "ERR_ZERO_ADDRESS");
+		address oldSOVTokenAddress = address(sovTokenAddress);
+		sovTokenAddress = newSOVTokenAddress;
+
+		emit SetSOVTokenAddress(msg.sender, oldSOVTokenAddress, newSOVTokenAddress);
+	}
+
+	/**
+	 * @notice The feesController calls this function to withdraw fees
+	 * from sources: protocolFeeTokensHeld.
+	 * The fees will be converted to wRBTC
+	 *
+	 * @param receiver address which will received the protocol fee (would be feesController/feeSharingProxy)
+	 *
+	 * @return The withdrawn total amount in wRBTC
+	 * */
+	function withdrawFees(address receiver) external returns (uint256) {
+		require(msg.sender == feesController, "unauthorized");
+
+		uint256 amountConvertedToWRBTC;
+		uint256 tempAmountConvertedToWRBTC;
+		IERC20Token _token;
+		uint256 _tokenAmount;
+		IInternalSovrynSwapNetwork sovrynSwapNetwork = IInternalSovrynSwapNetwork(addressOf(SOVRYNSWAP_NETWORK));
+
+		for (uint256 i = 0; i < reserveTokens.length; i++) {
+			_token = IERC20Token(reserveTokens[i]);
+			_tokenAmount = protocolFeeTokensHeld[address(_token)];
+			if(_tokenAmount <= 0) continue;
+
+			protocolFeeTokensHeld[address(_token)] = 0;
+
+			if (_token == sovTokenAddress) {
+				_token.approve(feesController, _tokenAmount);
+				IFeeSharingProxy(feesController).transferTokens(sovTokenAddress, uint96(_tokenAmount));
+				tempAmountConvertedToWRBTC = 0;
+			} else {
+				if (_token == wrbtcAddress) {
+					tempAmountConvertedToWRBTC = _tokenAmount;
+				} else {
+					bool successOfApprove = _token.approve(addressOf(SOVRYNSWAP_NETWORK), _tokenAmount);
+					require(successOfApprove, "ERR_APPROVAL_FAILED");
+
+					IERC20Token[] memory path = sovrynSwapNetwork.conversionPath(_token, IERC20Token(wrbtcAddress));
+
+					uint256 minReturn = sovrynSwapNetwork.rateByPath(path, _tokenAmount).mul(995).div(1000);
+
+					tempAmountConvertedToWRBTC = sovrynSwapNetwork.convert(
+						path,
+						_tokenAmount,
+						minReturn
+					);
```

Comment:
this slippage protection is ineffective, since `minReturn` is calculated within transaction with actual swap. Consider either
- replacing `minReturn`  with `0`, which saves gas
- or getting from the source different from  `sovrynSwapNetwork`, e.g., as a transaction parameter or from any other oracle

Reply:
Used 1 instead of 0